src/prog_models/sim_result.py

```diff
@@ -126,13 +145,11 @@ def to_numpy(self, keys = None) -> np.ndarray:
         """
         if len(self.data) == 0:
             return np.array([[]], dtype=np.float64)
-        if len(self.data[0]) == 0:
-            return np.array([[] for _ in self.data], dtype=np.float64)
-        if isinstance(self.data[0], DictLikeMatrixWrapper) and keys is None:
-            return np.array([u_i.matrix[:, 0] for u_i in self.data], dtype=np.float64)
-        if keys is None:
-            keys = self.data[0].keys()
-        return np.array([[u_i[key] for key in keys] for u_i in self.data], dtype=np.float64)
+        if keys is not None:
+            if all(item in self.frame.columns.to_list() for item in keys) is True:
+                with_keys_numpy = self.frame.drop(['time'], axis=1)[keys].to_numpy(dtype=np.float64)
+                return with_keys_numpy
```

Comment:
We discussed this over teams and decided to keep the generator version for speed. The below change keeps this while implementing the other changes. Namely, 
* Raises an exception if there are missing keys
* removes the unnecessary "is True"
* removes the unnecessary drop

```suggestion
        if keys is not None:
            if not all(item in self.frame.columns.to_list() for item in keys):
                raise KeyError('Every key in keys must be present in SimResult.')
            return self.frame[keys].to_numpy(dtype=np.float64)
```